### PR TITLE
[FLINK-27626] Introduce pre-aggregated merge to table store

### PR DESCRIPTION
**Pull request by HannanKan**
We can introduce richer merge strategies, one of which is already introduced is PartialUpdateMergeFunction, which completes non-NULL fields when merging. This PR introduces support for pre-aggregated merges.

### CODE REVIEW

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldAggregator.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    break;
+                default:
+                    throw new ValidationException(
+                            "Use unsupported aggregation or spell aggregate function incorrectly!");
+            }
+        }
+        return fieldAggregator;
+    }
+
+    abstract void agg(
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldAggregator.java:76]: Why not just `Object agg(Object accumulator, Object inputField)`?

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldMinAgg.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            int fieldPos) {
+        Object min;
+        Object mergeField = fieldGetters[fieldPos].getFieldOrNull(mergeRow);
+        Object inField = fieldGetters[fieldPos].getFieldOrNull(inRow);
+
+        if (mergeField == null || inField == null) {
+            min = (mergeField == null ? inField : mergeField);
+        } else {
+            // ordered by type root definition
+            switch (fieldType.getTypeRoot()) {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldMinAgg.java:48]: Can we just extract a method `compare` to reuse logical for min and max? Maybe just in `RowDataUtils`.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldPKAgg.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+package org.apache.flink.table.store.file.mergetree.compact.aggregate;
+
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.LogicalType;
+
+/** aggregate by primary keys, so no need to change them except initializing. */
+public class FieldPKAgg extends FieldAggregator {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldPKAgg.java:26]: Maybe just use `LAST_VALUE`？

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/RowAggregator.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.types.logical.LogicalType;
+
+import java.io.Serializable;
+import java.util.List;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/** Provide an Aggregator for merge a new row data. */
+public class RowAggregator implements Serializable {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/RowAggregator.java:30]: Maybe just merge this class into `AggregateMergeFunction`?

**Review by JingsongLi — COMMENTED** [review]
Thanks @HannanKan for the contribution.
Left some comments.

**Review by JingsongLi — COMMENTED** [review]
Hi @HannanKan , can you add more unit tests to functions? Now there is only itcase.

**Comment by HannanKan** [comment]
> Hi @HannanKan , can you add more unit tests to functions? Now there is only itcase.

Good idea! I will do it later.

**Comment by HannanKan** [comment]
> Hi @HannanKan , can you add more unit tests to functions? Now there is only itcase.

I have added uint testings for key functions and updated docs. I appreciate if you can review the code again.

**Comment by HannanKan** [comment]
@JingsongLi @alpinegizmo @aljoscha @rmetzger  Excuse me. I found that my pr need approve to run github action each time I update branch related to pr. I wonder whether this step could be removed so that we work more efficiently?

**Review by JingsongLi — APPROVED** [review]
Thanks @HannanKan for the update!
Looks good to me!

**Pull request merged**